cinema/models.py

```diff
@@ -41,10 +50,73 @@ def __str__(self):
 
 
 class Seat(models.Model):
-
     hall = models.ForeignKey(CinemaHall, on_delete=models.CASCADE, related_name='seats')
     code = models.CharField(max_length=4, default='1A')
     occupied = models.BooleanField(default=False)
 
     def __str__(self):
         return f'Seat {self.code} in hall {str(self.hall)}'
+
+
+class Contact(models.Model):
+    name = models.CharField(max_length=40)
+    email = models.EmailField()
+    phone = PhoneNumberField()
+    city = models.CharField(max_length=20, blank=True, null=True)
+    cinema = models.CharField(max_length=30, blank=True, null=True)
+    subject = models.CharField(max_length=100)
+    message = models.CharField(max_length=1500)
+
+    def __str__(self):
+        return self.email
+
+
+class Schedule(models.Model):
+    movie = models.ForeignKey(Movie, on_delete=models.CASCADE)
+    cinema = models.ForeignKey(Cinema, on_delete=models.CASCADE)
+    hall = ChainedForeignKey(
+        CinemaHall,
+        chained_field='cinema',
+        chained_model_field='cinema',
+        show_all=False,
+        auto_choose=True,
+        sort=True,
+    )
+    playing_time = models.DateTimeField()
+
+    @cached_property
+    def end_time(self):
+        end_time = self.playing_time
+        parsed_time = re.findall(r'\d+[ ]?', self.movie.length)
+
+        end_hour = end_time.hour + int(parsed_time[0])
+        end_minute = end_time.minute + int(parsed_time[1])
+
+        if end_minute > 59:
+            end_hour += 1
+            end_minute -= 60
+
+        if end_hour > 23:
+            end_time = self.playing_time + timedelta(days=1)
+            end_hour -= 24
+
+        end_time = end_time.replace(
+            hour=end_hour,
+            minute=end_minute,
+        )
+
+        return end_time
+
+    def clean(self):
+
+        for obj in Schedule.objects.all():
+            if (
+                    self.cinema.name == obj.cinema.name
+                    and self.hall.name == obj.hall.name
+                    and self.playing_time <= obj.end_time
+                    and self.end_time >= obj.playing_time
+            ):
```

Comment:
Not sure what this method is used for but you can do a filter instead of looping over all the existing objects and checking each obj value.
```suggestion
            ):Schedule.objects.filter(cinema__name=self.cinema.name, hall__name=self.hall.name, end_time=self.playing_time, playing_time=self.end_time)
            
```
Also, a unique_together (or even better, UniqueConstraint) might help to avoid having the same combination of cinema, hall, movie and playing_time in the DB.